File: solutions/javascript/sieve/1/sieve.js
//
// This is only a SKELETON file for the 'Sieve' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const primes = (n) => {
  let result = [];

  for (let i = 2; i <= n; i++) {
    let isPrime = true;

    for (let j = 2; j * j <= i; j++) {
      if (i % j === 0) {
        isPrime = false;
        break;
      }
    }

  
    if (isPrime) {
      result.push(i);
    }
  }

  return result;
};

File: solutions/python/sieve/1/sieve.py
def primes(limit):
    if limit < 2:
        return []

    is_prime = [True] * (limit + 1)
    is_prime[0] = False
    is_prime[1] = False

    
    for i in range(2, int(limit ** 0.5) + 1):
        if is_prime[i]:
            for multiple in range(i * i, limit + 1, i):
                is_prime[multiple] = False

    
    result = []
    for i in range(2, limit + 1):
        if is_prime[i]:
            result.append(i)

    return result
